go/vm/geth/geth.go

```diff
@@ -95,15 +96,92 @@ func (m *gethVm) Run(parameters vm.Parameters) (vm.Result, error) {
 	return vm.Result{}, fmt.Errorf("internal EVM error in geth: %v", err)
 }
 
+// TODO: remove once there is only one Revision definition
+func vmRevisionToCt(revision vm.Revision) ct.Revision {
+	switch revision {
+	case vm.R07_Istanbul:
+		return ct.R07_Istanbul
+	case vm.R09_Berlin:
+		return ct.R09_Berlin
+	case vm.R10_London:
+		return ct.R10_London
+	case vm.R11_Paris:
+		return ct.R11_Paris
+	case vm.R12_Shanghai:
+		return ct.R12_Shanghai
+	case vm.R13_Cancun:
+		return ct.R13_Cancun
+
+	}
+	panic(fmt.Sprintf("Unknown revision: %v", revision))
+}
+
+// makeChainConfig returns a chain config for the given chain ID and target revision.
+// The baseline config is used as a starting point, so that any prefilled configuration from go-ethereum params/config.go can be used.
+// chainId needs to be prefilled as it may be accessed with the opcode CHAINID.
+// the fork-blocks and the fork-times are set to the respective values for the given revision.
+func makeChainConfig(baseline params.ChainConfig, chainId *big.Int, targetRevision ct.Revision) params.ChainConfig {
+	istanbulBlock, err := ct.GetForkBlock(ct.R07_Istanbul)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to get Istanbul fork block: %v", err))
```

Comment:
nit: maybe a lambda func could be used to reproduce the same error msg just changing the revision.

Reply:
Yes, I agree, but I would rather leave this code for now, and clean it up when we can remove the errors from the getForkBlock and the two enums for revisions. 
Then, it is going to look much better.